src/NHibernate/Util/AsyncLock.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NHibernate.Util
+{
+	public class AsyncLock
+	{
+		private readonly AsyncSemaphore semaphore;
+		private readonly Task<Releaser> releaser;
```

Comment:
New code should follow the "underscore camelCase" convention. (NHibernate has lot of "camelCase only" private members due to ports from Java having forgotten to rename them, but we should avoid adding more.)
`_semaphore`, `_releaser`.